authorize/evaluator/evaluator.go

```diff
@@ -242,24 +219,60 @@ func (e *Evaluator) JWTPayload(req *Request) map[string]interface{} {
 
 	return payload
 }
+func newSigner(options *config.Options) (jose.Signer, *jose.JSONWebKey, error) {
+	// if we don't have a signing key, generate one
+	if options.SigningKey == "" {
+		key, err := cryptutil.NewSigningKey()
+		if err != nil {
+			return nil, nil, fmt.Errorf("couldn't generate signing key: %w", err)
+		}
+		generatedKey, err := cryptutil.EncodePrivateKey(key)
+		if err != nil {
+			return nil, nil, fmt.Errorf("bad signing key: %w", err)
+		}
+		options.SigningKey = base64.StdEncoding.EncodeToString(generatedKey)
+	}
+	if options.SigningKeyAlgorithm == "" {
+		options.SigningKeyAlgorithm = string(jose.ES256)
+	}
+
+	decodedCert, err := base64.StdEncoding.DecodeString(options.SigningKey)
+	if err != nil {
+		return nil, nil, fmt.Errorf("bad signing key: %w", err)
+	}
+	jwk, err := cryptutil.PrivateJWKFromBytes(decodedCert, jose.SignatureAlgorithm(options.SigningKeyAlgorithm))
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't generate signing key: %w", err)
+	}
+	log.Info().Str("Algorithm", jwk.Algorithm).
+		Str("KeyID", jwk.KeyID).
+		Interface("Public Key", jwk.Public()).
+		Msg("authorize: signing key")
 
-// SignedJWT returns the signature of given request.
-func (e *Evaluator) SignedJWT(payload map[string]interface{}) (string, error) {
 	signerOpt := &jose.SignerOptions{}
 	signer, err := jose.NewSigner(jose.SigningKey{
-		Algorithm: jose.ES256,
-		Key:       e.jwk,
-	}, signerOpt.WithHeader("kid", e.kid))
+		Algorithm: jose.SignatureAlgorithm(jwk.Algorithm),
+		Key:       jwk,
+	}, signerOpt.WithHeader("kid", jwk.KeyID))
+
 	if err != nil {
-		return "", err
+		return nil, nil, fmt.Errorf("couldn't create signer: %w", err)
+	}
+	return signer, jwk, nil
+}
+
+// SignedJWT returns the signature of given request.
+func (e *Evaluator) SignedJWT(payload map[string]interface{}) (string, error) {
+	if e.signer == nil {
```

Comment:
No longer sets up the signer on each evaluation which should improve performance.